azurerm/internal/services/batch/data_source_batch_pool.go

```diff
@@ -278,6 +278,14 @@ func dataSourceArmBatchPool() *schema.Resource {
 					},
 				},
 			},
+			"metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
```

Comment:
And won't need force new for a datasource

Reply:
removed Optional and ForeceNew